### PR TITLE
feat(TDP-5296): bootstrap react-cmf project

### PR DESCRIPTION
**Pull request by ncomont**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-5296

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [x] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/131/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/131/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/133/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/133/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/142/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/142/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/176/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/176/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/183/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/183/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/185/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/185/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/192/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/192/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/212/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/212/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/213/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/213/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/214/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/214/)

File: dataprep-dataset/src/main/resources/application.properties [hunk 1]
```diff
@@ -78,3 +78,4 @@ spring.output.ansi.enabled=always
 default.import.text.enclosure=\"
 default.import.text.escape=
 #################################
+
```
Comment (frassinier) [inline dataprep-dataset/src/main/resources/application.properties]: to be fixed

File: dataprep-webapp/config/sass.conf.js [hunk 2]
```diff
@@ -4,6 +4,19 @@ $brand-primary: #4F93A7;
 $brand-primary-t7: #00A1B3;
 $brand-secondary-t7: #168AA6;
 
+$tc-drawer-content-max-width: 100%;
+
+@mixin flex-full-height(){
```
Comment (frassinier) [inline dataprep-webapp/config/sass.conf.js]: Add comment to be fixed in dataset

File: dataprep-webapp/config/webpack.config.dev.js [hunk 3]
```diff
@@ -1,12 +1,47 @@
...
+config.devServer = {
+	port: 3000,
+	setup(app) {
+		app.get('/assets/config/config.json', function (req, res) {
+			appConfig.serverUrl = 'http://localhost:8888';
+			res.json(appConfig);
+		});
+	},
+	proxy: {
+		'/api/v1/stream-websocket': {
```
Comment (frassinier) [inline dataprep-webapp/config/webpack.config.dev.js]: add link to streams mock

File: dataprep-webapp/config/webpack.config.prod.js [hunk 4]
```diff
@@ -1,10 +1,27 @@
...
-		minify: true,
-		stripComments: true,
-	});
-};
+const webpack = require('webpack');
+const ReactCMFWebpackPlugin = require('@talend/react-cmf-webpack-plugin');
+
+const config = require('./webpack.config');
+
+config.devtool = 'source-map';
```
Comment (frassinier) [inline dataprep-webapp/config/webpack.config.prod.js]: @ncomont has a doubt

File: dataprep-webapp/karma.conf.js [hunk 5]
```diff
@@ -29,7 +29,7 @@ module.exports = function (config) {
 		webpack: webpackConfig,
 
 		webpackServer: {
-			noInfo: true // prevent console spamming when running in Karma!
+		//	noInfo: true // prevent console spamming when running in Karma!
```
Comment (frassinier) [inline dataprep-webapp/karma.conf.js]: to be un commented

File: dataprep-webapp/package.json [hunk 6]
```diff
@@ -11,24 +11,33 @@
     "clean": "rimraf ./build && rimraf ./coverage",
     "clean:build": "rimraf ./build",
     "clean:coverage": "rimraf ./coverage",
-    "build": "webpack --config config/webpack.config.dev.js --progress",
+    "build": "NODE_ENV=production webpack --config config/webpack.config.prod.js --progress",
```
Comment (frassinier) [inline dataprep-webapp/package.json]: build:dist is missing

File: dataprep-webapp/package.json [hunk 7]
```diff
@@ -11,24 +11,33 @@
...
     "prebuild": "rimraf ./build",
-    "build:dist": "webpack --config config/webpack.config.prod.js --progress",
-    "prebuild:dist": "rimraf ./build",
     "start": "npm run serve",
-    "serve": "webpack-dev-server --config config/webpack.config.dev.js --progress",
-    "serve:dist": "webpack-dev-server --config config/webpack.config.prod.js --progress --compress",
-    "test": "karma start",
-    "test:ci": "karma start karma.conf.ci.js",
-    "test:auto": "karma start --auto",
+    "serve": "NODE_ENV=development webpack-dev-server --config config/webpack.config.dev.js --progress --content-base build/",
```
Comment (frassinier) [inline dataprep-webapp/package.json]: serve:dist

File: dataprep-webapp/src/app/components/Version/Version.component.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+export default class Version extends React.PureComponent {
+
+	render() {
+		return (
+			<div>
+				<p><Icon name="talend-tdp-colored" /></p>
+				<dl className="dl-horizontal">
+					<dt>VERSION</dt>
+					Ò
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/Version/Version.component.js]: to be removed

File: dataprep-webapp/src/app/components/suggestions-stats/stats-details/stats-details-directive.spec.js [hunk 9]
```diff
@@ -81,27 +81,30 @@ describe('stats details directive', function () {
...
-		expect(element.find('.stat-table').eq(0).find('tr').eq(2).text().trim().replace(/ /g, '')).toBe('Duplicate:\n6');
-		expect(element.find('.stat-table').eq(0).find('tr').eq(3).text().trim().replace(/ /g, '')).toBe('Valid:\n9');
-		expect(element.find('.stat-table').eq(0).find('tr').eq(4).text().trim().replace(/ /g, '')).toBe('Empty:\n7');
-		expect(element.find('.stat-table').eq(0).find('tr').eq(5).text().trim().replace(/ /g, '')).toBe('Invalid:\n8');
-
-		expect(element.find('.stat-table').eq(1).find('tr').eq(0).text().trim().replace(/ /g, '')).toBe('MIN:\n10');
-		expect(element.find('.stat-table').eq(1).find('tr').eq(1).text().trim().replace(/ /g, '')).toBe('MAX:\n11');
-		expect(element.find('.stat-table').eq(1).find('tr').eq(2).text().trim().replace(/ /g, '')).toBe('Mean:\n12');
-		expect(element.find('.stat-table').eq(1).find('tr').eq(3).text().trim().replace(/ /g, '')).toBe('Variance:\n13');
+		const tables = element.find('.stat-table');
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/suggestions-stats/stats-details/stats-details-directive.spec.js]: to be checked

File: dataprep-webapp/src/app/configure.cmf.js [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+
+	return reduxLocalStorage
+		.loadInitialState({
+			key: LOCAL_STORAGE_KEY,
+			whitelist: [
+				['cmf', 'components', 'Container(SidePanel)'],
+				['cmf', 'components', 'Container(List)', 'preparations'],
+				['cmf', 'components', 'Container(List)', 'datasets'],
+				['cmf', 'components', 'Container(List)', 'datastores'],
+			],
```
Comment (frassinier) [inline dataprep-webapp/src/app/configure.cmf.js]: semantic ee

File: dataprep-webapp/src/app/index-module.js [hunk 11]
```diff
@@ -91,103 +90,93 @@ const app = angular
...
+		.module(SERVICES_UTILS_MODULE)
+		.value('copyRights', config.copyRights);
+}
+
+window.fetchConfiguration = getAppConfiguration;
 
+window.bootstrapDataPrepApplication = function bootstrapDataPrepApplication(modules, { config, appSettings }) {
+	appSettings.provider = appSettings.provider || 'legacy';
+
+	if (appSettings.provider.includes('catalog') && !(/playground/.test(window.location.href))) {
```
Comment (frassinier) [inline dataprep-webapp/src/app/index-module.js]: playround check must be improved

File: dataprep-webapp/src/app/index.themed.scss [hunk 12]
```diff
@@ -59,7 +59,7 @@
 		.vertical-barchart-cls .secondaryBar rect.bar {
 			fill: $brand-primary;
 		}
-
+		.tc-breadcrumb .tc-breadcrumb-item.active span,
```
Comment (frassinier) [inline dataprep-webapp/src/app/index.themed.scss]: wrong commit

File: dataprep-webapp/src/app/services/playground/playground-service.js [hunk 13]
```diff
@@ -1101,6 +1102,7 @@ export default function PlaygroundService(
 	function close() {
 		$timeout.cancel(fetchStatsTimeout);
 		$timeout(StateService.resetPlayground, 500, false);
-		$state.go(state.route.previous, state.route.previousOptions);
+		// $state.go(state.route.previous, state.route.previousOptions);
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/playground/playground-service.js]: check provider

File: dataprep-webapp/src/app/index-module.js (hunk 11 above)
Comment (frassinier) [reply dataprep-webapp/src/app/index-module.js]: add export check

File: dataprep-webapp/package.json [hunk 14]
```diff
@@ -11,24 +11,33 @@
...
+    "test": "NODE_ENV=test karma start && jest",
```
Comment (frassinier) [inline dataprep-webapp/package.json]: distinguish jasmine vs jest

**Pull request merged**